akka-stream/src/main/scala/akka/stream/impl/io/SslTls.scala

```diff
@@ -203,38 +203,76 @@ private[akka] class SslTlsCipherActor(settings: ActorFlowMaterializerSettings, s
    */
 
   var lastHandshakeStatus: HandshakeStatus = _
+  var corkUser = true
 
   val engineNeedsWrap = new TransferState {
     def isReady = lastHandshakeStatus == NEED_WRAP
-    def isCompleted = false
+    def isCompleted = engine.isOutboundDone
   }
 
   val engineInboundOpen = new TransferState {
-    def isReady = !engine.isInboundDone()
-    def isCompleted = false
+    def isReady = true
+    def isCompleted = engine.isInboundDone
```

Comment:
yes, this is the correct condition since inboundDone implies it cannot be ready again. (and no need to flip isReady anymore since isCompleted overrides it)